### PR TITLE
Add handleBatch implementation for PhpAmqpLib

### PR DESCRIPTION
**Pull request by rdrenth**
I've implemented batch handling using the [PhpAmqpLib batch methods](https://github.com/videlalvaro/php-amqplib/blob/master/demo/batch_publish.php).
The PHP AMQP extension does not support batch handling, so this is only implemented for PhpAmqpLib.

### CODE REVIEW

File: src/Monolog/Handler/AmqpHandler.php [hunk 1]
```diff
@@ -75,17 +69,72 @@ protected function write(array $record)
...
         }
+
+        $this->exchange->publish_batch();
+    }
+
+    /**
+     * Gets the routing key for the AMQP exchange
+     *
+     * @param array $record
+     * @returns string
```
Comment (stof) [inline src/Monolog/Handler/AmqpHandler.php]: typo: ``@return``

File: src/Monolog/Handler/AmqpHandler.php [hunk 2]
```diff
@@ -75,17 +69,72 @@ protected function write(array $record)
...
+     */
+    protected function getRoutingKey(array $record)
```
Comment (stof) [inline src/Monolog/Handler/AmqpHandler.php]: this should be private IMO

File: src/Monolog/Handler/AmqpHandler.php [hunk 3]
```diff
@@ -75,17 +69,72 @@ protected function write(array $record)
...
+        );
+
+        return strtolower($routingKey);
+    }
+
+    /**
+     * @param string $data
+     * @return AMQPMessage
+     */
+    protected function createAmqpMessage($data)
```
Comment (stof) [inline src/Monolog/Handler/AmqpHandler.php]: same here

**Comment by rdrenth** [comment]
@stof I've updated the PR according to you feedback. And yeah, making functions protected by default is an old habbit i need to get rid of :)

**Comment by Seldaek** [comment]
Thanks, I merged it in the 1.x  branch, see https://github.com/Seldaek/monolog/commit/0d2bef057998dd3294808d906aa6b3f71dc59439

**Comment by rdrenth** [comment]
Awesome, thanks!

**Pull request closed without merging**